### ISSUE TITLE
Feature Request. A Visual representation of My Templates

### ISSUE THREAD

**Issue by Flipback8** [pojome/elementor#3127]
is there any reason that My Templates cannot be displayed just like the Predesigned Templates section?It would be nice to visually see the template sections that I create.
--

**Comment by dudaster** [comment]
Also if you can put them in categories would be awesome.

**Comment by Flipback8** [comment]
Thanks. I will look into how to use Github. First time I visit that site.

**Comment by IDESIGNSTUDIO** [comment]
Oh, yes +2 for this one!
Very end-customer friendly.
Great UX7UI proposal

**Comment by Giffordcl** [comment]
With the addition of a hover over magnification, actually on all the libraries to get a larger view of the thumbnail without having to preview.

**Comment by pulla** [comment]
+1 
My template is not user friendly. inconvenience.

**Comment by devstrong** [comment]
+1
It's a must have feature request. Curious to know its progress.

**Comment by redlagoon** [comment]
Continuing the discussion here.

This is huge. While I do understand the fact that Elementor needs to get PRO subs to keep going (I'd pay you guys $1,000 for a license, great plugin), this will, for me, be a game breaker.

It means that I cannot use Elementor for any of my themes and I DON'T WANT to go back. Elementor's great, fast, but a way to add my own Templates' data, filters, previews is necessary.

I've built my own way to insert a template to a page, but it's out of Elementor, this has to be added. 

Not trying to be a fatalist, the world doesn't die, but dang, this is very useful.

**Comment by PwrSrg** [comment]
YES! YES! YES!  I can't believe there aren't a LOT more +1's in this thread.  I also can't believe this isn't one of the most popular feature requests.  Lastly, I am kinda surprised that 2.0 even released without this.  The "My Templates" feature is practically USELESS without it.  We currently have to create a public draft page in every one of our websites and insert every template individually just for a visual reference.  Very inefficient.  =(

**Comment by redlagoon** [comment]
@KingYes 

Any updates on this? Can you at least confirm you guys are not working on it as of now, so that we may work on a PR?

**Comment by iamntz** [comment]
Hey guys. So I did a bit of dig in this direction and here is what I found:

1. Templates _already_ supports thumbnails. Templates are basically post types and you can set a feature image on these. 
1. Templates also supports categories. There are only a bunch of categories though (Page, Section, Single and Widget) used internally.
1. The magic happens (or... rather _not_ happening) in the view. Because is a list view, thumbs are a bit harder to display. In order to fix this we need a grid view, similar to the blocks or pages template.

That being said: I could build it and submit a PR, but considering how open is pojome towards accepting PRs, I'm very demotivated to do so. :)

However, if you'd like fiddling around, you could use a combination of `wp_footer` and `elementor/editor/footer` hooks with output buffering, search & replace and completely override the view.

Adding 

```
	<# if (thumbnail) { #>
		<img src="{{{ thumbnail }}}" alt="">
	<# } #>
```

is a good start :)

https://github.com/pojome/elementor/blob/master/includes/editor-templates/templates.php#L174

**Comment by redlagoon** [comment]
@KingYes Can you please review @iamntz 's PR and approve of it?

**Comment by redlagoon** [comment]
@KingYes Can you please review @iamntz 's PR and approve of it?

**Comment by redlagoon** [comment]
@KingYes @bainternet Can you please review @iamntz 's updates and approve of it or give it green light?

**Comment by iamntz** [comment]
For those that are looking for a workaround, try this:

https://gist.github.com/30d3fddd6cd26db05a44a87605edc92a

It's all condensed into one function, for brevity, but you should at least split CSS to a separate file. This code will look like this: 

![](https://img.iamntz.com/2018-07-04_21h18_25.jpg)

You can either override the existing CSS (this would be tedious i guess) or try to rename all css classes to a different thing, to avoid clashing.

**Comment by bainternet** [comment]
@redlagoon 

We are working on something that makes a screen capture on template save, no ETA but its in the works.

**Comment by redlagoon** [comment]
@bainternet 

Amazing, please know that this is a feature any developer that's serious about selling his theme to more people **needs**, I'm in touch with people who've built successful themes with Elementor and they have users complaining about this.

Here's how VC does it:

https://www.youtube.com/watch?v=OtucEWLEoDI&t=26s

So, we need a featured image, a search and if possible, a filter for categories (such as buttons, hero images, etc.).

Thanks a ton and I can't wait to see it.

**Comment by davidmabelle** [comment]
Would love to see Templates be sortable, editable and viewable the way Admin Columns Pro does it. https://www.admincolumns.com/
Hoping someone makes a plugin for this if the Elementor team is too busy.

**Comment by made-man-12** [comment]
yes, we need this feature as soon as possible

**Comment by black-eye** [comment]
+1

**Comment by redlagoon** [comment]
I've been praying since at least 5 months ago.

I believe in you guys. 🙏

(Just bumping)

**Comment by juanpasolano** [comment]
I hope elementor does this, I think they know that we need this as much as they need it for their blocks. There's a reason they added the feature for them so I think It makes total sense for us dev to have it to.

**Comment by TheMangoTrain** [comment]
+1

**Comment by franciscorcc** [comment]
+1

**Comment by arielk** [comment]
ref: https://github.com/elementor/elementor/issues/2184

**Comment by hkmsadek** [comment]
Still nothing? How do happy adon, essential adon and other plugin does these things? You see a custom icon of their own and then you can simply install the templates easily from the custom template library

**Comment by kingfisher64** [comment]
Any updates on this please?

**Comment by bugnumber9** [comment]
It would be great to have this.
